### ISSUE TITLE
[HOLD for payment 2024-08-02] [HOLD for payment 2024-07-25] [$500] iOS - Room - Send message button stops functioning after leaving room

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#41780]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.71.0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Open the app and log in
2. Open a room or create one if you do not have any
3. Send a message
4. Tap the header > Leave
5. Return to the room
6. Send a message
7. Tap the Join button and try sending message again

## Expected Result:
The message is sent

## Actual Result:
The send button does not function, user can't send a message until the room is opened again

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/fe7065c2-6003-4b2b-9784-9f73ae442587

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~010a06a1bd2e5e09eb</li>
        <li>Upwork Job ID: 1789056197545811968</li>
        <li>Last Price Increase: 2024-07-10</li>
<li>Automatic offers: </li>
<ul>
<li>dominictb | Contributor | 103099455</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @johncschuster</details>

**Comment by lanitochka17** [comment]
@carlosmiceli  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp

**Comment by marcaaron** [comment]
I'm kind of curious about the expected behavior here.

>Tap the header > Leave
>Return to the room

If you left the room should you be able to return to it so easily by navigating back? Maybe it should clear from the navigation stack?

I think we can remove the Deploy Blocker label for this one. I find it unusual that someone would leave a room and want to leave a message right after (though agree - it looks like a clear bug).

@carlosmiceli maybe you want to confirm the expected behavior somewhere like #product or tag in Design team to get a second opinion.

**Comment by shawnborton** [comment]
Hmm I would think that if you leave a room, but you can still view the room, as soon as you send a message in that room you would re-join it. So I don't think we should block the send action from happening here. @Expensify/design what do you think?

**Comment by dubielzyk-expensify** [comment]
Don't we have a banner at the bottom for rooms you haven't joined? Or is that just for public rooms outside your workspace? Would it be a dumb idea to have something similar? So when leave a room it just pops up with that banner, but when you join it removes and the compose bar comes in?

If I take your question directly though I agree that we shouldn't block the send action. Though it makes you think what even is joining a room at that point? Is it just that it shows up in your LHN, then that's okay I guess?

A question answered with more questions :P I could've missed something glaring here

**Comment by carlosmiceli** [comment]
Trying to gather everyone's thoughts here: it seems that we just need to decide between:
- The user is able to go back to a room and message there after leaving it (therefore joining again). This is what the video shows that the user wanted to do.
- The user shouldn't be able to go back a room that they leave right after leaving by navigating back. If the user wanted to leave a message in that room again, the user would have to find/join the room the standard way. 

From my understanding of what @shawnborton is saying about why it should work like the first option, I lean towards that one and we should this as a bug that fixes/allows what the user is trying to do.

**Comment by dannymcclain** [comment]
@carlosmiceli Yeah I agree.

**Comment by shawnborton** [comment]
Yup, that's my understanding too.

**Comment by carlosmiceli** [comment]
Great, will proceed to fix it that way then, thanks everyone!

**Comment by carlosmiceli** [comment]
Trying to get to it this week, got sick last week and some things got pushed 🙏

**Comment by johncschuster** [comment]
@carlosmiceli I see you reapplied the `External` label. Does that mean I should be on the lookout for external contributors?

**Comment by carlosmiceli** [comment]
Hey @johncschuster, that's correct, but I thought that the message from Melvin above "Current assignee @thesahindia is eligible for the External assigner, not assigning anyone new." meant that that was happening automatically, my bad if I misunderstood that 🙏

**Comment by johncschuster** [comment]
Ah! Good shout. So, @thesahindia has been selected as the C+ reviewer, which is the first step when the `External` label is applied. At this point, we need to start hunting for proposals (assuming they don't naturally show up). Once We get a proposal we like, we'll assign the issue to an external contributor, at which point, Melvin will call out the two assignees on the issue and invite them to Upwork.

TL;DR - we're close, but we need proposals.

**Comment by carlosmiceli** [comment]
Right! I was hoping to see proposals coming in, is there anything I can/should help with to increase those coming in?

**Comment by marcaaron** [comment]
Nope. Just wait and review the proposals when @thesahindia recommends a good one.

**Comment by johncschuster** [comment]
[Raised in Slack to get some proposals](https://expensify.slack.com/archives/C01GTK53T8Q/p1718674826507989)

**Comment by shawnborton** [comment]
Not overdue, still waiting for proposals.

**Comment by johncschuster** [comment]
Acknowledged! Will issue payment after the regression window

**Comment by johncschuster** [comment]
@carlosmiceli I think ^that's you! Can you take a look at the linked PR to see if it caused a regression?

**Comment by johncschuster** [comment]
@thesahindia can you complete the BZ Checklist? Once complete, go ahead and request payment!

**Comment by johncschuster** [comment]
# Payment Summary:
Contributor: @dominictb paid $500 via [Upwork](https://www.upwork.com/nx/wm/offer/103099455) - PAID 🎉 
Contributor+: @thesahindia owed $500 via NewDot

**Comment by johncschuster** [comment]
Thank you for calling that out, @thesahindia! You are correct!

@dominictb I accidentally issued the non-penalized amount of $500 via Upwork earlier. I've just sent a request for a refund for $250. I've also adjusted the payment summary above to reflect the payment amounts in light of the regression.

**Comment by johncschuster** [comment]
Thank you, @dominictb ❤️

**Comment by johncschuster** [comment]
Looks like the last step here is for @thesahindia to request payment. The [payment summary](https://github.com/Expensify/App/issues/41780#issuecomment-2265459588) is up to date. Please go ahead and request payment!

**Comment by JmillsExpensify** [comment]
$250 approved for @thesahindia
